sdk/communication/azure-communication-administration/tests/test_phone_number_administration_client_async.py

```diff
@@ -0,0 +1,285 @@
+        self.phonenumber_for_capabilities = os.getenv('AZURE_COMMUNICATION_SERVICE_TNM_PHONENUMBER_FOR_CAPABILITIES', "+17771234567")
+        self.capabilities_id = os.getenv('AZURE_COMMUNICATION_SERVICE_TNM_CAPABILITIES_ID', "capabilities-id-1")
+        self.release_id = os.getenv('AZURE_COMMUNICATION_SERVICE_TNM_RELEASE_ID', "release_id_1")
+        self.scrubber.register_name_pair(
+            self.phone_plan_group_id,
+            "phone_plan_group_id"
+        )
+        self.scrubber.register_name_pair(
+            self.phone_plan_id,
+            "phone_plan_id"
+        )
+        self.scrubber.register_name_pair(
+            self.phone_plan_id_area_codes,
+            "phone_plan_id_area_codes"
+        )
+        self.scrubber.register_name_pair(
+            self.area_code_for_search,
+            "area_code_for_search"
+        )
+        self.scrubber.register_name_pair(
+            self.search_id,
+            "search_id"
+        )
+        self.scrubber.register_name_pair(
+            self.search_id_to_purchase,
+            "search_id_to_purchase"
+        )
+        self.scrubber.register_name_pair(
+            self.search_id_to_cancel,
+            "search_id_to_cancel"
+        )
+        self.scrubber.register_name_pair(
+            self.phonenumber_to_configure,
+            "phonenumber_to_configure"
+        )
+        self.scrubber.register_name_pair(
+            self.phonenumber_to_get_config,
+            "phonenumber_to_get_config"
+        )
+        self.scrubber.register_name_pair(
+            self.phonenumber_to_unconfigure,
+            "phonenumber_to_unconfigure"
+        )
+        self.scrubber.register_name_pair(
+            self.phonenumber_for_capabilities,
+            "phonenumber_for_capabilities"
+        )
+        self.scrubber.register_name_pair(
+            self.capabilities_id,
+            "capabilities_id"
+        )
+        self.scrubber.register_name_pair(
+            self.release_id,
+            "release_id"
+        )
+
+    @AsyncCommunicationTestCase.await_prepared_test
+    @pytest.mark.live_test_only
+    async def test_list_all_phone_numbers(self):
+        async with self._phone_number_administration_client:
+            pages = self._phone_number_administration_client.list_all_phone_numbers()
+
+            items = []
+            async for item in pages:
+                items.append(item)
+        self.assertGreater(len(items), 0)
+
+    @AsyncCommunicationTestCase.await_prepared_test
+    @pytest.mark.live_test_only
+    async def test_get_all_area_codes(self):
+        async with self._phone_number_administration_client:
+            area_codes = await self._phone_number_administration_client.get_all_area_codes(
+                location_type="NotRequired",
+                country_code=self.country_code,
+                phone_plan_id=self.phone_plan_id_area_codes
+            )
+        assert area_codes.primary_area_codes is not None
```

Comment:
Given primary_area_codes is not boolean, you can just use 
assert area_codes.primary_area_codes